src/UI/Component/MessageBox/Factory.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+/* Copyright (c) 2018 Thomas Famula <famula@leifos.de> Extended GPL, see docs/LICENSE */
+
+namespace ILIAS\UI\Component\MessageBox;
+
+/**
+ * This is how a factory for Message Boxes looks like.
+ */
+interface Factory {
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The system failed to complete some actions and displays information about the failure.
+	 *   composition: >
+	 *     The div uses the alert-danger Bootstrap style.
+	 *
+	 * rules:
+	 *   usage:
+	 *      1: >
+	 *          The Failure Message Boxes MUST be used, if a user interaction has failed.
+	 *      2: >
+	 *          The message SHOULD inform the user why the interaction has failed.
+	 *      3: >
+	 *          The message SHOULD inform the user how to the problem can be fixed.
+	 * ---
+	 *
+	 * @return \ILIAS\UI\Component\MessageBox\MessageBox
+	 */
+	public function failure($message_text);
+
+	/**
+	 * ---
+	 * description:
+	 *   purpose: >
+	 *     The system succeeded in finishing some action and displays a success message.
+	 *   composition: >
+	 *     The div uses the alert-success Bootstrap style.
```

Comment:
Very technical, consider refrasing "the alert-warning color is used as background of the message."